src/strategy/strategy.ts

```diff
@@ -28,11 +31,33 @@ export class Strategy {
     this.returnValue = returnValue;
   }
 
+  
+
   checkConstraint(constraint: Constraint, context: Context) {
     const field = constraint.contextName;
     const contextValue = resolveContextValue(context, field);
-    const isIn = constraint.values.some((val) => val.trim() === contextValue);
-    return constraint.operator === Operator.IN ? isIn : !isIn;
+    if(!contextValue) {
+      return false;
+    }
+    switch (constraint.operator) {
+      case Operator.IN:
+      case Operator.NOT_IN: {
+        const isIn = constraint.values.some((val) => val.trim() === contextValue);
+        return constraint.operator === Operator.IN ? isIn : !isIn;
+      }
+      case Operator.ENDS_WITH: {
+        return constraint.values.some((val) => contextValue.endsWith(val.trim()));
+      }
+      case Operator.STARTS_WITH: {
+        return constraint.values.some((val) => contextValue.startsWith(val.trim()));
+      }
+      case Operator.CONTAINS: {
+        return constraint.values.some((val) => contextValue.includes(val.trim()));
+      }
+      default: 
+        return false;
+    }
```

Comment:
Instead of a switch on the Operator-enum I am tempted to instead create a small handler function per operator and just execute that function. 

Example for contains:

```
const handleContains = (values: string[], contextValue) => values.some((val) => contextValue.includes(val.trim()))
```

Reply:
Also, we should probably pre-map the values by trimming them, making the handler work on pure data. 

```ts
values.map((v) => v.trim())
````